tests/tests/tier0/bluechi-daemon-reload/test_bluechi_deamon_reload.py

```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import logging
+
+from typing import Dict
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+
+LOGGER = logging.getLogger(__name__)
+NODE_FOO = "node-foo"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+    service_file = "dont-perform-deamon-reload.service"
+    service_path = os.path.join("/", "etc", "systemd", "system", service_file)
+
+    node_foo.copy_systemd_service(service_file, "systemd", os.path.join("/", "etc", "systemd", "system"))
+    ctrl.bluechictl.start_unit(NODE_FOO, service_file, expected_result=0)
+    assert node_foo.wait_for_unit_state_to_be(service_file, "failed", delay=2)
+
+    node_foo.exec_run(f"sed -i '/ExecStart=/c\\ExecStart=/bin/true' {service_path}")
+    ctrl.bluechictl.daemon_reload_node(NODE_FOO)
+    ctrl.bluechictl.start_unit(NODE_FOO, service_file)
+    assert node_foo.wait_for_unit_state_to_be(service_file, "active", delay=2)
```

Comment:
Steps in this test don't seem to aligned with the test goal, I'd suggest following steps:

1. Install simple.service with copy_systemd_services and daemon-reload disable
2. Run `bluechictl status node-foo simple.service`, which should return an error about unknown service
3. Run `bluechictl dameon-reload node-foo`
4. Wait until simple.service is inactive

Reply:
The problem I encountered after installing the service is that I can see the service as failed in `bluechictl status node-foo simple.service` and not an  error

Reply:
That is why I don't use the simple.service and made my own  service that suppose to fail at first

Reply:
Hmm, it seems that systemd is watching file changes in `/etc/systemd/system` so daemon-reload is not necessary at all (at least on current CS9):

1. `simple.service` doesn't exists as a file on mperina-cs9-2
   ```
   [root@mperina-cs9 ~]# bluechictl status mperina-cs9-2 simple.service
UNIT		| LOADED	| ACTIVE	| SUBSTATE	| FREEZERSTATE	| ENABLED	|
------------------------------------------------------------------------------------------------
Unit simple.service could not be found.
[root@mperina-cs9 ~]# echo $?
0
   ```
2. Create `/etc/systemd/system/simple.service`:
   ```
[root@mperina-cs9 ~]# bluechictl status mperina-cs9-2 simple.service
UNIT		| LOADED	| ACTIVE	| SUBSTATE	| FREEZERSTATE	| ENABLED	|
------------------------------------------------------------------------------------------------
simple.service	| loaded	| inactive	| dead		| running	| disabled	|
[root@mperina-cs9 ~]# echo $?
0   
   ```
3. Run `systemctl deamon-reload` on node mperina-cs9-2
   ```
[root@mperina-cs9 ~]# bluechictl status mperina-cs9-2 simple.service
UNIT		| LOADED	| ACTIVE	| SUBSTATE	| FREEZERSTATE	| ENABLED	|
------------------------------------------------------------------------------------------------
simple.service	| loaded	| inactive	| dead		| running	| disabled	|
[root@mperina-cs9 ~]# echo $?
0
   ```
4. Remove `/etc/systemd/system/simple.service`:
   ```
   [root@mperina-cs9 ~]# bluechictl status mperina-cs9-2 simple.service
UNIT		| LOADED	| ACTIVE	| SUBSTATE	| FREEZERSTATE	| ENABLED	|
------------------------------------------------------------------------------------------------
Unit simple.service could not be found.
[root@mperina-cs9 ~]# echo $?
0
   ```

So we will need to figure out a bit different scenario to make sure that bluechictl daemon-reload works as expected

Reply:
I don't think so. Because if I will comment out the daemon-reload line the service will still be failed.

Reply:
That is why I intentionally make the service fail and the only way to make it active is to change it and the daemon-reload.

Reply:
As far as I know, systemd caches units. That implies that a previously removed unit (e.g. simple.service) - removing the unit file and daemon-reload - will have the previous state when a unit with the same name is created again. 

So the steps @mwperina suggests should be the way to go with a few changes:
- don't use `copy_systemd_files` and use `create_file` instead (target to `/etc/systemd/system`)
- define a plain and valid service (with `ExecStart=/bin/true`) from the start - no need to make it fail first

Reply:
I don't think caching is involved here. I created `simple123456.service`, which I'm 100% sure I haven't used it before, and systemd knows about it as the same time it was created in `/etc/systemd/systemd`.

Reply:
I tried Michael's approach but it didn't help. the service is in `inactive` status right after the copy

Reply:
Hmm... interesting. Then it seems to only apply to changes of the units (according to [daemon-reload](https://www.freedesktop.org/software/systemd/man/latest/systemctl.html#daemon-reload)). 
In that case lets stick to this test flow. And please revert the changes in `machine.py`.